pkg/function/backup_data.go

```diff
@@ -64,13 +68,19 @@ func init() {
 
 var _ kanister.Func = (*backupDataFunc)(nil)
 
-type backupDataFunc struct{}
+type backupDataFunc struct {
+	progressPercent string
+}
 
 func (*backupDataFunc) Name() string {
 	return BackupDataFuncName
 }
 
-func (*backupDataFunc) Exec(ctx context.Context, tp param.TemplateParams, args map[string]interface{}) (map[string]interface{}, error) {
+func (b *backupDataFunc) Exec(ctx context.Context, tp param.TemplateParams, args map[string]interface{}) (map[string]interface{}, error) {
+	// Set progress percent
+	b.progressPercent = progress.PercentProgressStarted
+	defer func() { b.progressPercent = progress.PercentProgressCompleted }()
```

Comment:
would we want to set the progress as completed even if the function execution failed?

Reply:
No, for failed phase, the progress will be nil. I've added the sample in the test plan
```
status:                                                                                                                                                                                                     
  actions:                                                                                                                                                                                                            
  - artifacts:                                                                                                                                                                                                        
      operatorBackup:                                                                                                                                                                                                 
        keyValue:                                                                                                                                                                                                     
          path: '{{ .Phases.phase3.Output.path }}'                                                                                                                                                                    
    blueprint: operator-blueprint                                                                                                                                                                                     
    deferPhase:                                                                                                                                                                                                       
      name: ""                                                                                                                                                                                                        
      progress: {}                                                                                                                                                                                                    
      state: ""                                                                                                                                                                                                       
    name: backup                                                                                                                                                                                                      
    object:                                                                                                                                                                                                           
      apiVersion: v1                                                                                                                                                                                                  
      group: apps                                                                                                                                                                                                     
      kind: statefulset                                                                                                                                                                                               
      name: mysql                                                                                                                                                                                                     
      namespace: mysql                                                                                                                                                                                                
      resource: statefulsets                                                                                                                                                                                          
    phases:                                                                                                                                                                                                           
    - name: phase1                                                                                                                                                                                                    
      output:                                                                                                                                                                                                         
        path: /operator-backups/2023-05-22T07-32-36/backup.tgz                                                                                                                                                        
      progress:                                                                                                                                                                                                       
        lastTransitionTime: "2023-05-22T07:32:54Z"                                                                                                                                                                    
        progressPercent: "100"                                                                                                                                                                                         
      state: complete                                                                                                                                                                                                 
    - name: phase2                                                                                                                                                                                                    
      progress: {}                                                                                                                                                                                                    
      state: failed                                                                                                                                                                                                   
    - name: phase3                                                                                                                                                                                                    
      progress: {}                                                                                                                                                                                                    
      state: pending                                                                                                                                                                                                  
  error:                                                                                                                                                                                                              
    message: "Failed while waiting for Pod kanister-job-sk2bk to complete: Pod failed                                                                                                                                 
      or did not transition into complete state: Running backup\
     .
     .
  progress:                                                                                                                                                             
    lastTransitionTime: "2023-05-22T07:32:54Z"                                          
    percentCompleted: "33"                                                           
  state: failed  
```